### PR TITLE
Add restore old GUID

### PR DESCRIPTION
**Pull request by Mmduh-483**

### CODE REVIEW

File: pkg/sriov/sriov.go [hunk 1]
```diff
@@ -70,6 +70,32 @@ func (p *pciUtilsImpl) GetPciAddress(ifName string, vf int) (string, error) {
...
+	for _, vfObj := range pfHandle.List {
+		if vfObj.PciAddress == vfPciAddress {
+			vf = vfObj
+			found = true
+		}
+	}
+	if !found {
+		return fmt.Errorf("failed to find VF %s for PF %s", vfPciAddress, pfName)
+	}
+	if err = sriovnet.UnbindVf(pfHandle, vf); err != nil {
```
Comment (adrianchiris) [inline pkg/sriov/sriov.go]: in general is kinda wierd that you need the PF name to unbind/bind the VF from the driver.

for example for mellanox nics using mlx5 driver you just:
echo [VF PCI D:B:D.F address] > /sys/bus/pci/drivers/mlx5_core/unbind
echo [VF PCI D:B:D.F address] > /sys/bus/pci/drivers/mlx5_core/bind

File: pkg/sriov/sriov.go [hunk 2]
```diff
@@ -255,6 +223,36 @@ func (s *sriovManager) ApplyVFConfig(conf *types.NetConf) error {
 		}
 	}
 
+	// Set link guid
+	if conf.GUID != "" {
+		if len(conf.GUID) < 22 || conf.GUID == "00:00:00:00:00:00:00:00" {
```
Comment (moshe010) [inline pkg/sriov/sriov.go]: I think is better to create method to validate GUID by regex

File: pkg/sriov/sriov.go [hunk 3]
```diff
@@ -276,5 +274,27 @@ func (s *sriovManager) ResetVFConfig(conf *types.NetConf) error {
 		}
 	}
 
+	// Reset link guid
+	if conf.HostIFGUID != "" {
+		if conf.HostIFGUID == "00:00:00:00:00:00:00:00" {
```
Comment (moshe010) [inline pkg/sriov/sriov.go]: put a comment on why this is done

File: pkg/sriov/sriov.go [hunk 4]
```diff
@@ -276,5 +274,27 @@ func (s *sriovManager) ResetVFConfig(conf *types.NetConf) error {
...
+		if err = s.nLink.LinkSetVfNodeGUID(pfLink, conf.VFID, guid); err != nil {
+			return fmt.Errorf("failed to add node guid %s: %v", guid, err)
+		}
+		if err = s.nLink.LinkSetVfPortGUID(pfLink, conf.VFID, guid); err != nil {
+			return fmt.Errorf("failed to add port guid %s: %v", guid, err)
+		}
+		// unbind vf then bind it to apply the guid
+		if err = s.utils.RebindVf(conf.Master, conf.DeviceID); err != nil {
+			return err
+		}
```
Comment (moshe010) [inline pkg/sriov/sriov.go]: L 283 to 294 look like duplicate code to Line 229

File: pkg/sriov/sriov_test.go [hunk 5]
```diff
@@ -31,6 +32,72 @@ func (l *FakeLink) Type() string {
...
 
+	Context("Checking ApplyVFConfig function", func() {
+		var (
+			netconf *types.NetConf
+		)
+
+		BeforeEach(func() {
+			netconf = &types.NetConf{
+				Master:      "i4",
+				DeviceID:    "0000:af:06.0",
```
Comment (moshe010) [inline pkg/sriov/sriov_test.go]: Add invalid tests case according to regex
such as 
1. invalid char
2. wrong lenght 
...

File: pkg/sriov/sriov.go (hunk 2 above)
Comment (Mmduh-483) [reply pkg/sriov/sriov.go]: added

File: pkg/sriov/sriov.go (hunk 3 above)
Comment (Mmduh-483) [reply pkg/sriov/sriov.go]: added

File: pkg/sriov/sriov.go (hunk 4 above)
Comment (Mmduh-483) [reply pkg/sriov/sriov.go]: add `setVfGUID` function to sriovManger struct to avoid code duplication

File: pkg/sriov/sriov.go (hunk 1 above)
Comment (Mmduh-483) [reply pkg/sriov/sriov.go]: Opened issue for it https://github.com/Mellanox/sriovnet/issues/26

File: pkg/sriov/sriov_test.go (hunk 5 above)
Comment (Mmduh-483) [reply pkg/sriov/sriov_test.go]: added

**Pull request merged**